File: php/features.php
<?php
declare(strict_types=1);

// ErrorHandling SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class ErrorHandlingFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new ErrorHandlingBaseFeature();
            case "test":
                return new ErrorHandlingTestFeature();
            default:
                return new ErrorHandlingBaseFeature();
        }
    }
}

File: py/features.py
# ErrorHandling SDK feature factory

from feature.base_feature import ErrorHandlingBaseFeature
from feature.test_feature import ErrorHandlingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ErrorHandlingBaseFeature(),
        "test": lambda: ErrorHandlingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
